File: 981-delete-columns-to-make-sorted/delete-columns-to-make-sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n = strs[0].size();
        unordered_set<int> s;
        for(int i=0;i<n;i++)
        {
            for(int j=1;j<strs.size();j++)
            {
                if(strs[j][i]<strs[j-1][i])
                {
                    s.insert(i);
                    break;
                }
            }
        }

        return s.size();
    }
};

File: 981-delete-columns-to-make-sorted/delete-columns-to-make-sorted.java
class Solution {
    public int minDeletionSize(String[] strs) {
        int n = strs[0].length();
        Set<Integer> st = new HashSet<>();
        for(int i=0;i<n;i++)
        {
            boolean flag = false;
            for(int j=1;j<strs.length;j++)
            {
                if(strs[j].charAt(i)<strs[j-1].charAt(i)) {
                    flag = true;
                    break;
                }
            }

            if(flag)st.add(i);
        }

        return st.size();
    }
}